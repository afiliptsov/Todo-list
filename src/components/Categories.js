import React from 'react'
import {FiDelete} from 'react-icons/fi'
import '../styles/Categories.css'

const Categories = props => {
  const {categoriesList, toggleCategory, addCategory, removeCategory} = props
  return (
    <div className="categoriesWrapper">
      <ul>
        {categoriesList.map(category => (
          <li
            className={
              category.active ? 'listCategory activeCategory' : 'listCategory'
            }
            // style={category.active ? {color: 'red'} : {color: 'blue'}}
            onClick={() => toggleCategory(category.id)}
            key={category.id}
          >
            {category.key}
            <FiDelete className="icon" />{' '}
          </li>
        ))}
      </ul>
    </div>
  )
}

export default Categories
