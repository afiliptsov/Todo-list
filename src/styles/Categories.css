.categoriesWrapper {
  min-width: 100px;
  max-width: 200px;
  margin-right: 10px;
  border-right: 1px grey solid;
  padding-right: 5px;
}
.categorySection {
  display: flex;
  cursor: pointer;
  border: 1px;
}
.categorySectionActive {
  background-color: #7e7e7e;
  border: 1px solid #b5b5b5;
}
.listCategory {
  width: 100%;
  padding: 2px;
}
.activeCategory {
  color: white;
}

.icon {
  padding: 2px;
  vertical-align: middle;
  margin-right: 2px;
  margin-left: 2px;
  float: right;
}
.iconActive {
  color: white;
}
