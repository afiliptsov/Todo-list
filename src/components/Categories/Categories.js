import React from 'react'
import {FiDelete} from 'react-icons/fi'
import '../../styles/Categories.css'
import PropTypes from 'prop-types'

const Categories = props => {
  const {categoriesList, toggleCategory, removeCategory} = props
  return (
    <div className="categoriesWrapper" data-testid="categoriesComponent">
      <ul>
        {categoriesList.map(category => (
          <div key={category.id} className="categorySection">
            <li
              onClick={() => toggleCategory(category.id)}
              className={
                category.active ? 'listCategory activeCategory' : 'listCategory'
              }
              key={category.id}
            >
              <span data-testid={category.key}>{category.key}</span>
            </li>
            <div data-testid="delete">
              <FiDelete
                onClick={() => removeCategory(category.id)}
                className="icon"
              />
            </div>
          </div>
        ))}
      </ul>
    </div>
  )
}

//props type checking
Categories.propTypes = {
  categoriesList: PropTypes.array,
  categoriestoggleCategoryList: PropTypes.func,
  removeCategory: PropTypes.func,
}

export default Categories
