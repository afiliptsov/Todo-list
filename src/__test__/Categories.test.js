import {render, screen} from '@testing-library/react'
import Categories from '../components/Categories/Categories'
import mockState from '../state/mockState'
import renderer from 'react-test-renderer'

describe('Categories page elements are displayed', () => {
  it('renders Categories  items component', () => {
    render(<Categories categoriesList={mockState} />)
    const getCategoryComponent = screen.getByTestId('categoriesComponent')
    expect(getCategoryComponent).toBeInTheDocument()
  })
  it('intro is displayed from mock state', () => {
    render(<Categories categoriesList={mockState} />)
    const getCategoryComponent = screen.getByTestId('intro')
    expect(getCategoryComponent.innerHTML).toBe('intro')
  })
})

describe('Categories component snapshot validation', () => {
  it('categories should match snapshot', () => {
    const component = renderer
      .create(<Categories categoriesList={mockState} />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
