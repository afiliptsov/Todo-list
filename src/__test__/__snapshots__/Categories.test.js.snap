// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component snapshot validation categories should match snapshot 1`] = `
<div
  className="categoriesWrapper"
  data-testid="categoriesComponent"
>
  <ul>
    <div
      className="categorySection"
    >
      <li
        className="listCategory activeCategory"
        onClick={[Function]}
      >
        <span
          data-testid="intro"
        >
          intro
        </span>
      </li>
      <div
        data-testid="delete"
      >
        <svg
          className="icon"
          fill="none"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 4H8l-7 8 7 8h13a2 2 0 0 0 2-2V6a2 2 0 0 0-2-2z"
          />
          <line
            x1="18"
            x2="12"
            y1="9"
            y2="15"
          />
          <line
            x1="12"
            x2="18"
            y1="9"
            y2="15"
          />
        </svg>
      </div>
    </div>
    <div
      className="categorySection"
    >
      <li
        className="listCategory"
        onClick={[Function]}
      >
        <span
          data-testid="groceries"
        >
          groceries
        </span>
      </li>
      <div
        data-testid="delete"
      >
        <svg
          className="icon"
          fill="none"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 4H8l-7 8 7 8h13a2 2 0 0 0 2-2V6a2 2 0 0 0-2-2z"
          />
          <line
            x1="18"
            x2="12"
            y1="9"
            y2="15"
          />
          <line
            x1="12"
            x2="18"
            y1="9"
            y2="15"
          />
        </svg>
      </div>
    </div>
  </ul>
</div>
`;
